contracts/contracts/poolBooster/PoolBoosterMerkl.sol

```diff
@@ -0,0 +1,125 @@
+        bytes32 _hashToSign,
+        address _creator,
+        bytes memory _campaignData
+    ) {
+        require(_rewardToken != address(0), "Invalid rewardToken address");
+        require(
+            _merklDistributor != address(0),
+            "Invalid merklDistributor address"
+        );
+        require(_hashToSign != bytes32(0), "Invalid hashToSign address");
+        require(_campaignData.length > 0, "Invalid campaignData");
+        require(_duration > 1 hours, "Invalid duration");
+
+        campaignType = _campaignType;
+        duration = _duration;
+        hashToSign = _hashToSign;
+        creator = _creator;
+
+        merklDistributor = IMerklDistributor(_merklDistributor);
+        rewardToken = IERC20(_rewardToken);
+        campaignData = _campaignData;
+    }
+
+    /// @notice Create a campaign on the Merkl distributor
+    function bribe() external override {
+        // Ensure token is approved for the Merkl distributor
+        uint256 minAmount = merklDistributor.rewardTokenMinAmounts(
+            address(rewardToken)
+        );
+        require(minAmount > 0, "Min reward amount must be > 0");
+
+        // if balance too small or below threshhold, do no bribes
+        uint256 balance = rewardToken.balanceOf(address(this));
+        if (
+            balance < MIN_BRIBE_AMOUNT ||
+            (balance * 1 hours < minAmount * duration)
+        ) {
+            return;
+        }
+
+        // Approve the bribe contract to spend the reward token
+        rewardToken.approve(address(merklDistributor), balance);
+
+        // Notify the bribe contract of the reward amount
+        merklDistributor.signAndCreateCampaign(
+            IMerklDistributor.CampaignParameters({
+                campaignId: bytes32(0),
+                creator: creator,
+                rewardToken: address(rewardToken),
+                amount: balance,
+                campaignType: campaignType,
+                startTimestamp: getTomorrowRoundedTimestamp(),
+                duration: duration,
+                campaignData: campaignData
+            }),
+            bytes("")
+        );
+        emit BribeExecuted(balance);
+    }
+
+    /// @notice Used to sign a campaign on the Merkl distributor
+    /// @param hash Hash of the data to be signed
+    function isValidSignature(bytes32 hash, bytes memory)
+        external
+        view
+        override
+        returns (bytes4 magicValue)
+    {
+        // Check if the signature is valid for the given hash
+        // bytes4(keccak256("isValidSignature(bytes32,bytes)")) == 0x1626ba7e
+        return (hash == hashToSign ? bytes4(0x1626ba7e) : bytes4(0x00000000));
+    }
+
+    /// @notice Returns the current timestamp rounded to the next day
+    function getTomorrowRoundedTimestamp() public view returns (uint32) {
+        // Calculate the timestamp for the start of tomorrow (next midnight)
+        return uint32((block.timestamp / 1 days + 1) * 1 days);
```

Comment:
I think this needs to use the `duration` and not `1 days`. What if the `duration` was 7 days? Wouldn't that mean we can get overlapping campaigns?

The alternative is to fix the `duration` to 1 day.

Reply:
Nice catch thanks! 
Indeed I should have use duration and not days and I prefer to keep duration flexible.
Fixed in this commit: [7b20bfe](https://github.com/OriginProtocol/origin-dollar/pull/2474/commits/7b20bfeb1cb5788047f686b840e6c8bccccb7bdf)
I changed the function name to have something more meaningful.